autopush/websocket.py

```diff
@@ -657,28 +672,24 @@ def _send_ping(self):
         return self.sendMessage("{}", False)
 
     def process_ping(self):
-        """Adaptive ping processing
+        """Ping Handling
 
         Clients in the wild have a bug that lowers their ping interval to 0. It
-        will never increase for them, but if we disconnect them, then they will
-        reconnect in 5 seconds. As such, its beneficial for us and the client
-        to delay a response to a client pinging fast, but not such that its
-        worse than the alternative, reconnecting.
-
-        Therefore we will attempt to send this ping within the 10 second
-        timeout many clients already have, based on guesstimating latency from
-        the last ping. The last ping is not necessarilly latency, but it will
-        allow us to avoid sending too fast, but not waiting too long. The
-        practical result is that we will respond to each ping within 0-9 sec
-        depending on when we last got a ping.
+        will never increase for them, as there is no way to remedy this without
+        causing the client to use drastically more battery/data-usage we send
+        them a code 4774 close to signify that they should stop until network
+        change.
+
+        No other client should ping more than once per minute, or we tell them
+        to go away.
 
         """
         now = time.time()
         last_ping_ago = now - self.last_ping
-        if last_ping_ago >= 9:
+        if last_ping_ago >= 55:
             self._send_ping()
         else:
-            return self.deferToLater(9 - last_ping_ago, self._send_ping)
+            self.sendClose(code=4774)
```

Comment:
Actually, could you add a counter here, please? I'm wondering how many times we'll hit this...